From 171742ba0dc16918aaaf7db2b6b67991578b6525 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Tue, 30 Jul 2013 14:17:19 +0800
Subject: [PATCH 35/39] ENGR00273070 Enable 3G data connection for JB4.3

Make query_ctec return failure for HUAWEI&AMAZON modem
Since they dont support it

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 reference-ril/reference-ril.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/reference-ril/reference-ril.c b/reference-ril/reference-ril.c
index 9840535..2c13378 100755
--- a/reference-ril/reference-ril.c
+++ b/reference-ril/reference-ril.c
@@ -3955,7 +3955,7 @@ int query_ctec(ModemInfo *mdm, int *current, int32_t *preferred)
 
     if ((HUAWEI_MODEM == modem_type) ||
             (AMAZON_MODEM == modem_type)){
-        return 0;
+        return -1;
     }else{
         RLOGD("query_ctec. current: %d, preferred: %d", (int)current, (int) preferred);
         err = at_send_command_singleline("AT+CTEC?", "+CTEC:", &response);
-- 
1.8.0

